sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -808,6 +824,45 @@ object SymmetricHashJoinStateManager {
     result
   }
 
+  def mergeStateStoreCheckpointInfo(
```

Comment:
I already commented about this elsewhere (that it shouldn't be in the symmetric hash join state manager), but this was confusing to read because it is used in two places:

1. To merge the key with index state store with the key with index to value state store
2. To merge the results from step (1) for both the left and the right into one result